webpack/stripes-module-parser.js

```diff
@@ -36,9 +39,26 @@ class StripesModuleParser {
 
   // Wrapper to source data and transform into collections (config and metadata)
   parseModule() {
+    const stripes = this.packageJson.stripes;
+    if (!_.isObject(stripes)) {
+      throw new StripesBuildError(`Included module ${this.moduleName} does not have a "stripes" key in package.json`);
+    }
+
+    // Upgrade string actsAs to singleton array and provide compatibility for deprecated options
+    let actsAs = stripes.actsAs;
+    if (!Array.isArray(actsAs)) {
+      if (typeof actsAs === 'string') actsAs = [actsAs];
+      else if (typeof stripes.type === 'string') {
```

Comment:
Looks like this is still backwards compatible with `stripes.type`, nice!  Would we want to maintain at least one test to verify that `stripes.type` is still supported (until some future major version when its not)?

Also this may be a good place to log/warn that folks should upgrade their package.json to `actAs`.

Reply:
Thanks Matt! You're right, a deprecation warning is in order. Added some.

I think the stripes.type -> actsAs compatibility code is pretty explicitly that and not in danger of being inadvertently removed so I'm comfortable not adding a test for it. While I very much think tests are well worth maintaining I think there is a point of diminishing returns and am more skeptical of the super granular ones. By that token I also didn't include a check to ensure each of the deprecation warnings is appropriately triggered.